bindings/java/src/main/native/jucx_common_def.cc

```diff
@@ -16,14 +19,82 @@ static inline void log_error(const char* error)
  */
 JNIEXPORT void JNICALL JNU_ThrowException(JNIEnv *env, const char *msg)
 {
-  jclass cls = env->FindClass("org/ucx/jucx/UcxException");
-  log_error(msg);
-  if (cls != 0) {/* Otherwise an exception has already been thrown */
-    env->ThrowNew(cls, msg);
-  }
+    jclass cls = env->FindClass("org/ucx/jucx/UcxException");
+    log_error(msg);
+    if (cls != 0) { /* Otherwise an exception has already been thrown */
+        env->ThrowNew(cls, msg);
+    }
 }
 
 void JNU_ThrowExceptionByStatus(JNIEnv *env, ucs_status_t status)
 {
     JNU_ThrowException(env, ucs_status_string(status));
 }
+
+bool j2cInetSockAddr(JNIEnv *env, jobject sock_addr, sockaddr_storage& ss,  socklen_t& sa_len)
+{
+    memset(&ss, 0, sizeof(ss));
+    sa_len = 0;
+
+    if (sock_addr == NULL)
+    {
+        JNU_ThrowException(env, "j2cInetSockAddr: InetSocketAddr is null");
+        return false;
+    }
+
+    jclass inetsockaddr_cls = env->GetObjectClass(sock_addr);
+
+    // Get sockAddr->port
+    jmethodID getPort = env->GetMethodID(inetsockaddr_cls, "getPort", "()I");
+    jint port = env->CallIntMethod(sock_addr, getPort);
+
+    // Get sockAddr->getAddress (InetAddress)
+    jmethodID getAddress = env->GetMethodID(inetsockaddr_cls, "getAddress",
+                                            "()Ljava/net/InetAddress;");
+    jobject inet_address = env->CallObjectMethod(sock_addr, getAddress);
+
+    if (inet_address == NULL)
+    {
+        JNU_ThrowException(env, "j2cInetSockAddr: InetSocketAddr.getAddress is null");
+        return false;
+    }
+
+    jclass inetaddr_cls = env->GetObjectClass(inet_address);
+
+    // Get the byte array that stores the IP address bytes in the InetAddress.
+    jmethodID get_addr_bytes = env->GetMethodID(inetaddr_cls, "getAddress", "()[B");
+    jobject ip_byte_array = env->CallObjectMethod(inet_address, get_addr_bytes);
+
+    if (ip_byte_array == NULL) {
+        JNU_ThrowException(env, "j2cInetSockAddr: InetAddr.getAddress.getAddress is null");
+        return false;
+    }
+    jbyteArray addressBytes = static_cast<jbyteArray>(ip_byte_array);
+    jsize addrlen = env->GetArrayLength(addressBytes);
+
+    if (addrlen == 4)
+    {
+        // Deal with Inet4Address instances.
+        // We should represent this Inet4Address as an IPv4 sockaddr_in.
+        ss.ss_family = AF_INET;
+        sockaddr_in &sin = reinterpret_cast<sockaddr_in &>(ss);
+        sin.sin_port = htons(port);
+        jbyte *dst = reinterpret_cast<jbyte *>(&sin.sin_addr.s_addr);
+        env->GetByteArrayRegion(addressBytes, 0, 4, dst);
+        sa_len = sizeof(sockaddr_in);
+        return true;
+    } else  {
```

Comment:
need to check if addrlen==16?
also can we check by address family instead of address length?